akka-actor/src/main/scala/akka/event/LoggerMailbox.scala

```diff
@@ -43,6 +43,5 @@ private[akka] class LoggerMailbox(owner: ActorRef, system: ActorSystem)
         envelope = dequeue
       }
     }
-    super.cleanUp(owner, deadLetters)
```

Comment:
@patriknw if you call super when the logger is down, super delivers to dead letters and dead letters sends to logger, if there are messages between end of loop and call to super then you are really creating an endless loop hence saying that somehow messages are still being received hence making that call not harmless anymore.